src/ring_theory/prime.lean

```diff
@@ -15,22 +15,28 @@ open finset
 
 open_locale big_operators
 
+/-- If `x * y = a * ∏ i in s, p i` where `p i` is always prime, then
+  `x` and `y` can both be written as a divisor of `a` multiplied by
+  a product over a subset of `s`  -/
 lemma mul_eq_mul_prime_prod {α : Type*} [decidable_eq α] {x y a : R} {s : finset α}
   {p : α → R} (hp : ∀ i ∈ s, prime (p i)) (hx : x * y = a * ∏ i in s, p i) :
-  ∃ t u b c, t ∪ u = s ∧ disjoint t u ∧ b * c = a ∧
-    b * ∏ i in t, p i = x ∧ c * ∏ i in u, p i = y :=
+  ∃ (t u : finset α) (b c : R),
+      t ∪ u = s
+    ∧ disjoint t u
+    ∧ b * c = a
+    ∧ b * ∏ i in t, p i = x
+    ∧ c * ∏ i in u, p i = y :=
```

Comment:
```suggestion
      s = t ∪ u
    ∧ disjoint t u
    ∧ a = b * c
    ∧ x = b * ∏ i in t, p i
    ∧ y = c * ∏ i in u, p i :=
```
This reads more natural to me... but I'll leave it up to you.

Reply:
I don't think you can do `rintro rfl` with the equality that way around.

Reply:
I was wrong. I'll swap the order of the equality.